### PR TITLE
[Logs]: Use docker integration to collect logs by default on kubernetes.

### PR DESCRIPTION
**Pull request by ajacquemot**
### What does this PR do?

Use the docker integration by default to collect logs on kubernetes and added a config parameter to override the default logic.

### Motivation

We changed the logic in `6.11.2` but the kubernetes integration does not have all the features of the docker one.

### Additional Notes

https://github.com/DataDog/datadog-agent/pull/3340

### CODE REVIEW

File: pkg/logs/input/container/launcher.go [hunk 1]
```diff
@@ -18,27 +18,47 @@ import (
...
+// a kubernetes launcher if '/var/log/pods' is mounted ; this behaviour is reversed when
+// collectFromFiles is enabled.
+// Returns a noop launcher is none of those volumes are mounted.
+func NewLauncher(collectAll bool, collectFromFiles bool, sources *config.LogSources, services *service.Services, pipelineProvider pipeline.Provider, registry auditor.Registry) restart.Restartable {
+	var (
+		launcher restart.Restartable
+		err      error
+	)
+
+	if collectFromFiles {
```
Comment (mfpierre) [inline pkg/logs/input/container/launcher.go]: Given that you have early returns you could do:

```
if collectFromFile {
  try k8s
}
try docker
try k8s
```
?

Comment (ajacquemot) [reply pkg/logs/input/container/launcher.go]: I could, it's just a nits right ?

File: pkg/config/config.go [hunk 2]
```diff
@@ -352,6 +352,8 @@ func initConfig(config Config) {
 	config.BindEnvAndSetDefault("logs_config.open_files_limit", 100)
 	// add global processing rules that are applied on all logs
 	config.BindEnv("logs_config.processing_rules")
+	// enforce the agent to use files to collect container logs on kubernetes environment
+	config.BindEnvAndSetDefault("logs_config.k8s_container_use_file", false)
```
Comment (mfpierre) [inline pkg/config/config.go]: Is it the correct milestone? Looks like added feature 🤔

Comment (ajacquemot) [reply pkg/config/config.go]: this is a fix to a problem that has been introduced to `6.11.2` and which will arise much more in `6.12` if we don't do anything.

File: pkg/logs/input/container/launcher.go [hunk 3]
```diff
@@ -18,27 +18,47 @@ import (
...
-	kubernetesLauncher, err := kubernetes.NewLauncher(sources, services, collectAll)
-	if err == nil {
-		log.Info("Kubernetes launcher initialized")
-		return kubernetesLauncher
-	}
-	log.Infof("Could not setup the kubernetes launcher: %v", err)
+// By default returns a docker launcher if the docker socket is mounted and fallback to
+// a kubernetes launcher if '/var/log/pods' is mounted ; this behaviour is reversed when
+// collectFromFiles is enabled.
+// Returns a noop launcher is none of those volumes are mounted.
```
Comment (tt810) [inline pkg/logs/input/container/launcher.go]:
```suggestion
// Returns a noop launcher if none of those volumes are mounted.
```

File: pkg/logs/input/container/launcher.go (hunk 1 above)
Comment (ajacquemot) [reply pkg/logs/input/container/launcher.go]: As discussed IRL, I feel like it's less readable and I'll stick to code duplication here.

**Comment by tt810** [comment]
LGTM, @mfpierre your call.

File: releasenotes/notes/logs-container-configuration-toggle-f5f8b2831c6a575f.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# Each section from every releasenote are combined when the
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
```
Comment (albertvaka) [inline releasenotes/notes/logs-container-configuration-toggle-f5f8b2831c6a575f.yaml]: There is a better section for breaking changes: `upgrade`. These appear first in the release notes, and include all the things to pay attention to when upgrading to a newer version of the agent.

Comment (ajacquemot) [reply releasenotes/notes/logs-container-configuration-toggle-f5f8b2831c6a575f.yaml]: fixed

**Review by albertvaka — APPROVED** [review]

**Review by hush-hush — APPROVED** [review]

**Pull request merged**